### DISCUSSION TITLE
Undefined returned data when fetch function use Supabase (React Native)

### DISCUSSION THREAD

**Discussion by Khanhan2k2** [TanStack/query#5193] [Q&A] [answered]
Hi everybody, I am using react-native@0.66.5 and supabase to build an app, then I want to apply some library for supporting fetching and caching data, I found swr, and TanStack query. Unfortunately, i coudn't  use it.
I use supabase for fetch funcion:
![Screenshot 2023-03-27 105844](https://user-images.githubusercontent.com/100439816/227838249-e213ecdf-027f-446e-9dc4-3378ba34eabf.png)
Then I use that for useQuery
![Screenshot 2023-03-27 105905](https://user-images.githubusercontent.com/100439816/227838314-736b03ba-99fd-4691-8789-cf3c2bada94e.png)
But when I log it I recieve 
![Screenshot 2023-03-27 105931](https://user-images.githubusercontent.com/100439816/227838355-4129deba-3790-4f3e-b058-fb376501abe1.png)
Please help me,
Thankfully.

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

It looks like you're not returning anything from your test function.
